### PR TITLE
[SPARK-43032] [SS] [CONNECT] Python SQM bug fix

### PR DESCRIPTION
**Pull request by WweiL**
### What changes were proposed in this pull request?

Some bug fix for streaming ***connect*** python SQM

Note that I also changed ***non-connect***'s StreamingQueryManager `get()` API to return an `Optional[StreamingQuery]`.

Before it looks like this when you get a non-exist query:

```
>>> a = spark.streams.get("00000000-0000-0001-0000-000000000001") 
>>> a
<pyspark.sql.streaming.query.StreamingQuery object at 0x7f86465702b0>
>>> a.id
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/wei.liu/oss-spark/python/pyspark/sql/streaming/query.py", line 78, in id
    return self._jsq.id().toString()
AttributeError: 'NoneType' object has no attribute 'id'
```

But now it looks like:
```
>>> a = spark.streams.get("00000000-0000-0001-0000-000000000001") 
>>> a.id
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'id'
```
The only difference is the return type, which is not typically honored in Python... But not very sure if that's a breaking change

### Why are the changes needed?

Bug fix

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Manually tested. Also verified that it won't throw even without this fix so it's not that urgent

### CODE REVIEW

**Comment by WweiL** [comment]
@HyukjinKwon This is relatively small but changes something, can you take a look?
Also CC @rangadi @pengzhon-db

**Review by rangadi — APPROVED** [review]

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala [hunk 1]
```diff
@@ -2466,14 +2466,16 @@ class SparkConnectPlanner(val session: SparkSession) {
 
       case StreamingQueryManagerCommand.CommandCase.GET =>
         val query = session.streams.get(command.getGet)
```
Comment (rangadi) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:2468]: 'getGet` sounds off. May be we could rename to `get_query` (changing the name is ok, not an incompatible change at this stage).

**Review by rangadi — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
I'm fine because we understand `Connect` module is experimental.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**